<script crossorigin="anonymous"
  integrity="sha384-+jvb+jCJ37FkNjPyYLI3KJzQeD8pPFXUra3B/QJFqQ3txYrUPIP1eOfxK4h3cKZP"
  src="https://lib.baomitu.com/vue/2.6.11/vue.js"></script>
<pre>
  * fn in data 没有报错, 但是实际上 this 指向 Proxy, 不是 Vue 本身
  * data in methods 直接报错
</pre>
<div id="app">
  <button @click="set1()">切换 {{evt}}</button>

  <div @[evt]="DoAct" style="background:pink">操作区域 {{evt}}</div>
</div>
<script>
  window.kkk = new Vue({
    el: '#app',
    data() {
      return {
        evt: 'click',
        set1() {
          this.evt = this.evt === 'click' ? 'dblclick' : 'click'
          console.log(this) // Proxy 实例
        },
      }
    },
    methods: {
      // val: 'is',
      DoAct() {
        console.log(this.evt)
        console.log(this) // Vue 实例
      },
    },
  })
</script>
